src/leap/bitmask/core/dispatcher.py

```diff
@@ -0,0 +1,196 @@
+
+        subcmd = parts[1]
+        user, password = parts[2], parts[3]
+
+        bf = self._get_service('bonafide')
+
+        if subcmd == 'authenticate':
+            d = bf.do_authenticate(user, password)
+
+        elif subcmd == 'signup':
+            d = bf.do_signup(user, password)
+
+        elif subcmd == 'logout':
+            d = bf.do_logout(user, password)
+
+        elif subcmd == 'active':
+            d = bf.do_get_active_user()
+
+        d.addCallbacks(_format_result, _format_error)
+        return d
+
+    def do_EIP(self, *parts):
+        subcmd = parts[1]
+        eip_label = 'eip'
+
+        if subcmd == 'enable':
+            return _format_result(
+                self.core.do_enable_service(eip_label))
+
+        eip = self._get_service(eip_label)
+        if not eip:
+            return _format_result('eip: disabled')
+
+        if subcmd == 'status':
+            return _format_result(eip.do_status())
+
+        elif subcmd == 'disable':
+            return _format_result(
+                self.core.do_disable_service(eip_label))
+
+        elif subcmd == 'start':
+            # TODO --- attempt to get active provider
+            # TODO or catch the exception and send error
+            provider = parts[2]
+            d = eip.do_start(provider)
+            d.addCallbacks(_format_result, _format_error)
+            return d
+
+        elif subcmd == 'stop':
+            d = eip.do_stop()
+            d.addCallbacks(_format_result, _format_error)
+            return d
+
+    def do_MAIL(self, *parts):
+
+        subcmd = parts[1]
+        mail_label = 'mail'
+
+        if subcmd == 'enable':
+            return _format_result(
+                self.core.do_enable_service(mail_label))
+
+        m = self._get_service(mail_label)
+        bf = self._get_service('bonafide')
+
+        if not m:
+            return _format_result('mail: disabled')
+
+        if subcmd == 'status':
+            return _format_result(m.do_status())
+
+        elif subcmd == 'disable':
+            return _format_result(self.core.do_disable_service(mail_label))
+
+        elif subcmd == 'get_imap_token':
+            d = m.get_imap_token()
+            d.addCallbacks(_format_result, _format_error)
+            return d
+
+        elif subcmd == 'get_smtp_token':
+            d = m.get_smtp_token()
+            d.addCallbacks(_format_result, _format_error)
+            return d
+
+        elif subcmd == 'get_smtp_certificate':
+            # TODO move to mail service
+            # TODO should ask for confirmation? like --force or something,
+            # if we already have a valid one. or better just refuse if cert
+            # exists.
+            # TODO how should we pass the userid??
+            # - Keep an 'active' user in bonafide (last authenticated)
+            # (doing it now)
+            # - Get active user from Mail Service (maybe preferred?)
+            # - Have a command/method to set 'active' user.
```

Comment:
see the above comment, we'll have to revisit this when we add authentication for other dispatcher uses other than cli.